pkg/atomiccount/atomiccount_test.go

```diff
@@ -39,6 +39,9 @@ func (suite *AtomicCountUtilSuite) TestAdd(c *check.C) {
 
 	result := acount.Get()
 	c.Check(result, check.Equals, (int32)(12))
+
+	var nilAcount *AtomicInt
+	c.Check(nilAcount.Add(5), check.Equals, (int32)(0))
```

Comment:
It makes me confused why there is no panic when invoking the function of nil struct pointer. After a while, I understand: `nilAcount.Add(5)` equals `(*AtomicInt).Add(nilAcount, 5)` in GoLang, and the implementation of function `AtomicInt#Add` checks `nil` pointer before dereference it.

But is it a reasonable implementation?
* It's rare to check itself in a struct's own function. Why not let it panic directly?
* The behavior of the `AtomicInt` functions are inconsistent: `Set` doesn't check.
* Any value plus zero value should be equal to itself.

@Starnop WDYT